pmlc/compiler/compiler.cc

```diff
@@ -122,6 +123,10 @@ Executable::Executable(StringRef entry, StringRef target, ModuleOp programModule
   manager.addNestedPass<FuncOp>(createCanonicalizerPass());
   manager.addNestedPass<FuncOp>(createCSEPass());
 
+  manager.addPass(dialect::tile::createConstantTypesPass(floatx, intx));
+  manager.addNestedPass<FuncOp>(createCanonicalizerPass());
+  manager.addNestedPass<FuncOp>(createCSEPass());
```

Comment:
I think this belongs in the `builder.cc`, take a look at the `TileBuilder::MakeProgram` method, here is where takes a set of outputs to start with and traces down all the expressions that those outputs depend on. Then, it runs a pipeline to canonicalize, but just before canonicalization (which is presently where type inference is occuring), we'd want to update the types on the constant ops.

Reply:
removed these changes